massgen/tests/test_checklist_tools_server.py

```
@@ -0,0 +1,317 @@
+
+    mcp = fastmcp.FastMCP("test_checklist")
+    from massgen.mcp_tools.checklist_tools_server import _register_checklist_tool
+
+    _register_checklist_tool(mcp, specs_path)
+
+    # Extract the registered tool's handler
+    # FastMCP stores tools internally; we access the handler directly
+    for tool in mcp._tool_manager._tools.values():
+        if tool.name == "submit_checklist":
+            return tool.fn
+    raise RuntimeError("submit_checklist tool not found after registration")
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)